app/js/enhancedDate.js

```diff
@@ -1,5 +1,80 @@
 /* 
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ */
+var enhancedDate = (function enhancedDate() {
 
+    'use strict';
+
+    // Private data, not accessible outside the enhancedDate module
+    var data = {
```

Comment:
I would just declare two arrays here as separate variables. It will clean up the date.days and date.months. Figuring out when to use a configuration data object is more of a "what looks right" kind of decision. In the case of your days and months, it's clear from their context what they're for. If you have a lot of varied configuration data, having them on an object can help clarify that they're configuration data: config.interval, config.warning, etc.